File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/data/AuthManager.java
package technology.mainthread.apps.gatekeeper.data;

import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
import com.google.android.gms.common.api.GoogleApiClient;

import io.reactivex.Flowable;

public interface AuthManager {

    GoogleApiClient getGoogleApiClient();

    Flowable<Boolean> authWithFirebase(GoogleSignInAccount account);

    Flowable<Boolean> signOut();

    Flowable<Boolean> deleteAccount();

}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/data/AuthManager.kt
package technology.mainthread.apps.gatekeeper.data

import com.google.android.gms.auth.api.signin.GoogleSignInAccount
import com.google.android.gms.common.api.GoogleApiClient

import io.reactivex.Flowable

interface AuthManager {

    val googleApiClient: GoogleApiClient

    fun authWithFirebase(account: GoogleSignInAccount): Flowable<Boolean>

    fun signOut(): Flowable<Boolean>

    fun deleteAccount(): Flowable<Boolean>

}
